File: frontend/src/services/playerService.ts
import api from './api';
import type { Player } from '../types/player';

export const playerService = {
  /**
   * Get player profile with stats and match history
   */
  getPlayerById: async (playerId: string): Promise<Player> => {
    const response = await api.get(`/players/${playerId}`);
    return response.data;
  },
};

export default playerService;

File: backend/flaskr/services/player_service.py
"""Service layer for player-related business logic."""
from typing import Optional, Dict


class PlayerService:
    """Service for handling player data operations."""
    
    def __init__(self, db):
        """Initialize with database connection."""
        self.db = db
        self.players_collection = db.players
    
    def get_player_by_id(self, player_id: str) -> Optional[Dict]:
        """
        Fetch details for a specific player.
        
        Args:
            player_id: The unique identifier for the player
            
        Returns:
            Player details dictionary or None if not found
        """
        # Fetch player from MongoDB
        player = self.players_collection.find_one({'_id': player_id})
        
        return player
